.nuspec/Microsoft.Maui.Controls.SingleProject.targets

```diff
@@ -47,32 +47,41 @@
     <WindowsPackageType Condition="'$(EnablePreviewMsixTooling)' == 'true'">MSIX</WindowsPackageType>
   </PropertyGroup>
 
-  <!-- Removals -->
-  <ItemGroup Condition=" '$(EnableDefaultItems)' == 'true' and '$(SingleProject)' == 'true' ">
-    <Compile Remove="$(PlatformsProjectFolder)**/*$(DefaultLanguageSourceExtension)" />
-    <!-- Explicitly include the platform code -->
-    <Compile
-        Condition=" '$(TargetPlatformIdentifier)' == 'android' and '$(AndroidProjectFolder)' != '' "
-        Include="$(AndroidProjectFolder)**/*$(DefaultLanguageSourceExtension)" />
-    <Compile
-        Condition=" '$(TargetPlatformIdentifier)' == 'ios' and '$(iOSProjectFolder)' != '' "
-        Include="$(iOSProjectFolder)**/*$(DefaultLanguageSourceExtension)" />
-    <Compile
-        Condition=" '$(TargetPlatformIdentifier)' == 'maccatalyst' and '$(MacCatalystProjectFolder)' != '' "
-        Include="$(MacCatalystProjectFolder)**/*$(DefaultLanguageSourceExtension)" />
-    <Compile
-        Condition=" '$(TargetPlatformIdentifier)' == 'windows' and '$(WindowsProjectFolder)' != '' "
-        Include="$(WindowsProjectFolder)**/*$(DefaultLanguageSourceExtension)" />
+  <!--
+    Run before both _MauiInjectXamlCssAdditionalFiles and GenerateMSBuildEditorConfigFileShouldRun because
+    if for some reason the _MauiInjectXamlCssAdditionalFiles target is not run, we still get in at the
+    correct time for the build to be adjusted.
+  -->
+  <Target Name="_MauiRemovePlatformCompileItems"
+          BeforeTargets="_MauiInjectXamlCssAdditionalFiles;GenerateMSBuildEditorConfigFileShouldRun"
```

Comment:
At some point we might change `GenerateMSBuildEditorConfigFileShouldRun` -> `CoreCompile` everywhere, but let's do that later if this is working. Should we file an issue on that?

Reply:
Done! https://github.com/dotnet/maui/issues/4431